Seda/Seda/Controllers/PaymentViewController.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  PaymentViewController.swift
+//  Seda
+//
+//  Created by Billy Chen on 5/17/20.
+//  Copyright © 2020 Billy Chen. All rights reserved.
+//
+
+import UIKit
+import Stripe
+import CardScan
+
+class PaymentViewController: UIViewController {
+    @IBOutlet weak var cardTextField: STPPaymentCardTextField!
+    @IBOutlet weak var cardNumberLabel: UILabel!
+    @IBOutlet weak var cashAmount: UITextField!
+    
+    let backendUrl = "http://localhost:4242/"
+    var scanStats: ScanStats?
+    var number: String?
+    var expiration: String?
+    var name: String?
+    var cardImage: UIImage?
+    var cashToSend: String = "0.00"
+    var paymentIntentClientSecret: String?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        print("viewDidLoad")
+        startCheckout()
+        //self.cardNumberLabel.text = format(number: self.number ?? "")
+        // Do any additional setup after loading the view.
+    }
+    
+    @IBAction func payPressed(_ sender: UIButton) {
+        pay()
+    }
+    
+    func format(number: String) -> String {
+        if number.count == 16 {
+            return format16(number: number)
+        } else if number.count == 15 {
+            return format15(number: number)
+        } else {
+            return number
+        }
+    }
+    
+    func format15(number: String) -> String {
+        var displayNumber = ""
+        for (idx, char) in number.enumerated() {
+            if idx == 4 || idx == 10 {
+                displayNumber += " "
+            }
+            displayNumber += String(char)
+        }
+        return displayNumber
+    }
+    
+    func format16(number: String) -> String {
+        var displayNumber = ""
+        for (idx, char) in number.enumerated() {
+            if (idx % 4) == 0 && idx != 0 {
+                displayNumber += " "
+            }
+            displayNumber += String(char)
+        }
+        return displayNumber
+    }
+    
+    func startCheckout() {
+        // Request a PaymentIntent from your server and store its client secret
+        // Create a PaymentIntent by calling the sample server's /create-payment-intent endpoint.
+        
+        let url = URL(string: backendUrl + "create-payment-intent")!
+        let json: [String: Any] = [
+          "items": [
+              ["id": "CASHTOSEND"]//TODO: CASHTOSEND should be taken from the UITextField, checked that it's a valid amount, and sent to our server
```

Comment:
In the comment are you saying that this amount needs to be verified with the user's balance on Firebase and then updated as well?

Reply:
Yea pretty much. This comment was just to remind myself or anyone who ends up implementing this that we don't want to just send some raw user input without verifying the user has that amount and that it's actually a number since we get it from a UITextLabel, so it's a string. Now that I think about it there's probably an input field that only takes in numbers.